version: '3.7'
services:

  download:
    container_name: tdm_download
    build:
      context: .
      dockerfile: Dockerfile/Dockerfile_donwload
    environment:
      - KAGGLE_USERNAME=verniereloic
      - KAGGLE_KEY=03e78f24e94d43d80b595934fc372ecf
    restart: on-failure
    volumes:
      - shared-data:/share
    networks:
      - net_common
    depends_on:
      - rabbitmq

  acquisition:
    container_name: tdm_acquisition
    build:
      context: .
      dockerfile: Dockerfile/Dockerfile_acquisition
    restart: on-failure
    volumes:
      - shared-data:/share
    networks:
      - net_common
    depends_on:
      - rabbitmq

  recommendation:
    container_name: tdm_recommandation
    build:
      context: .
      dockerfile: Dockerfile/Dockerfile_recommendation
    restart: on-failure
    volumes:
      - shared-data:/share
    networks:
      - net_common
    depends_on:
      - rabbitmq

  rabbitmq:
    container_name: tdm_rabbitmq
    image: 'rabbitmq:3.6-management-alpine'
    ports:
      - '5672:5672'
      - '15672:15672'
    healthcheck:
        test: ["CMD", "curl", "-f", "http://localhost:15672"]
        interval: 30s
        timeout: 10s
        retries: 5
    networks:
      - net_common

  notebook:
    container_name: tdm_notebooklab
    image: jupyter/datascience-notebook
    ports:
      - "8888:8888"
    volumes:
      - ./src/:/home/jovyan
      - shared-data:/share
    environment:
      - JUPYTER_ENABLE_LAB=yes
    command: start-notebook.sh --NotebookApp.token=''

volumes:
  shared-data:

networks:
  net_common:
    external:
      name: net_common
